commands/store.js

```diff
@@ -21,99 +20,85 @@ class Store extends Social {
+          if (['y', 'yes'].includes(response.toLowerCase())) {
+          
+            await connection.query('BEGIN');
+            const deducted = await this.client.db.takeUserCurrency(connection, message.guild.id, message.author.id, storeItem.value);
+            if (!deducted) {
+              await connection.query('ROLLBACK');
+              return message.channel.send('You don\'t appear to have the funds for that.');
+            }
+            await this.client.db.giveUserItem(connection, message.guild.id, message.author.id, storeItem.id, 1);
+            await connection.query('COMMIT');
+            return message.channel.send('You have bought the item :tada:');
+          } else
+          
+          if (['n', 'no', 'cancel'].includes(response.toLowerCase())) {
+            return message.channel.send('Transaction cancelled.');
+          }
+        } finally {
+          connection.release();
         }
         break;
       }
 
       case ('sell'): {
         const name = args.join(' ');
-        
-        if (!name) return this.client.commands.get('store').run(message, args, level);
-            
-        const item = this.client.store.filter(i => i.name.toLowerCase().includes(name.toLowerCase()));
-              
-        if (!item) return message.channel.send('That item doesn\'t exist, Please make sure it is spelled correctly');
-        
-        const returnPrice = Math.floor(item.array()[0].price/2);
-        
-        const response = await this.client.awaitReply(message, `Are you sure you want to sell ${item.array()[0].name} for <:blobcoin:${settings.blobCoin}>${returnPrice}?`, undefined, null);
-        if (['y', 'yes'].includes(response.toLowerCase())) {
-        
-          message.member.givePoints(returnPrice);
-          message.channel.send('You have sold the item :tada: ');
-        
-        } else
-        
-        if (['n', 'no', 'cancel'].includes(response.toLowerCase())) {
-          message.channel.send('Transaction cancelled.');
-        }
-        break;
-      }
 
-      case ('add'): {
-        const price = args.pop();
-        const name = args.join(' ');
+        const connection = await this.client.db.acquire();
+        let storeItem;
+        try {
+          storeItem = await this.client.db.getStoreItemByName(connection, name);
 
-        if (!name) return message.reply('Please add the exact name of the item');
-        
-        if (this.client.store.has(name)) return message.reply('This item is already on sale');
-        
-        if (!price) return message.reply('Please specify a price');
-        
-        const item = { name: name.toLowerCase(), id: Math.random(), price: price, guildId: message.guild.id };
-        this.client.store.set(item.id, item);
-        message.reply(`${name} is now on sale `);
-        break;
-      }
+          const returnPrice = Math.floor(storeItem.value/2);
```

Comment:
Perhaps make this a config option?

Reply:
I believe it is fair to return half the price. If requested by more Police, it can be changed to a config option.